src/RedisTaggedCache.js

```diff
@@ -105,6 +106,89 @@ class RedisTaggedCache extends TaggedCache {
       });
   }
 
+  /**
+   * Batch delete cache data belonging to a set of tags. This method performs
+   * deletions incrementally as tag set members are found (using sscan)
+   * instead of collecting all members for all tags and then deleting them
+   * simultaneously.
+   *
+   * {@link https://redis.io/commands/scan}
+   */
+  async batchDeleteWithTags(): Promise<Array<() => void>> {
+    const tagIds = await this.tags.tagIds();
+    return Promise.all(
+      _.map(tagIds, tagId =>
+        this.trampoline(
+          this.batchDeleteTagMembers.bind(this, this.referenceKey(tagId))
+        )
+      )
+    );
+  }
+
+  /**
+   * Takes a tag ID and deletes all associated members in batches.
+   *
+   * This method is intended to be called using the this.trampoline method,
+   * as it deletes the current batch, and returns a function that will delete
+   * the next batch until there are no more members in the tag set.
+   *
+   * @param {string} tagRef
+   *   String representing the reference key of the tag for which
+   *   members should be deleted.
+   * @param {number} cursor
+   *   Number representing the pagination cursor from the redis sscan call.
+   *
+   * @return {function|null}
+   *   Returns a function that can be called to delete the next batch of
+   *   members, or null if no more members exist.
+   *
+   * {@link https://redis.io/commands/scan}
+   */
+  async batchDeleteTagMembers(
+    tagRef: string,
+    cursor: number = 0
+  ): Promise<?() => any> {
+    // Fetch one iteration of tag members for the given tag id.
+    const { cursor: newCursor, members }: MemberPage = await this.getMemberPage(
+      tagRef,
+      cursor
+    );
+
+    // Process given keys.
+    const keysToDelete = _.map(members, key =>
+      key.replace(new RegExp(`^${this.store.options.keyPrefix}`), '')
+    );
+
+    await Promise.all([
+      // Remove all keys from the tag set.
+      ..._.map(keysToDelete, key => this.store.srem(tagRef, key)),
```

Comment:
Does it make sense to control the concurrency on this? For very large tag sets this would result in a lot of parallel requests to redis...

If your store is using a connection pool that'd be one way to control it, but alternatively we could use something like bluebird, or just

```javascript
_.chunk(keysToDelete, CONCURRENCY_LIMIT).reduce(async (last, chunk) => {
  await last;
  return Promise.all(chunk.map(key => this.store.srem(tagRef, key)));
}, Promise.resolve());
```

Reply:
I'll look into the size of the batches we're getting from redis, but I think they're pretty small. Regardless, controlling the concurrency makes sense.

Reply:
We discussed this in Slack:

> I'm not seeing any batches that contain more than 10 items. If I'm reading this right, the default count for scan calls is 10: https://redis.io/commands/scan#the-count-option